mempool/src/shared_mempool.rs

```diff
@@ -168,11 +168,17 @@ pub struct TransactionExclusion {
 
 /// Mempool's subscription to reconfiguration events
 pub struct MempoolReconfigSubscription {
-    sender: libra_channel::Sender<(), ContractEvent>,
+    sender: libra_channel::Sender<(), OnChainConfigPayload>,
 }
 
 impl EventSubscription for MempoolReconfigSubscription {
-    fn publish(&mut self, payload: ContractEvent) {
+    fn subscribed_configs(&self) -> HashSet<ConfigID> {
+        vec![VMPublishingOption::CONFIG_ID]
```

Comment:
do you need the vec? I thought slice works too

Reply:
with slice we need an extra clone, even with `into_iter` :/